### ISSUE TITLE
MongoDB syncs are failing and causing CPU overload

### ISSUE THREAD

**Issue by VikramTiwari** [metabase/metabase#7191]
Mongo based tables are failing on sync so badly. Maybe it's because of the unknown or mismatch types for the same key. As much as that is common in Mongo, the sync ops don't seem to be built to handle it.

- Your databases: MongoDB
- Metabase version: 0.28.2
- Metabase internal database: MySQL

Logs:
```
03-23 07:00:21 ERROR sync.util :: Error syncing fields for Table 2614 'abc': class java.lang.NullPointerException
("driver.mongo$describe_table_field.invokeStatic(mongo.clj:123)"
 "driver.mongo$describe_table_field.invoke(mongo.clj:120)"
 "driver.mongo$describe_table_field$iter__45688__45692$fn__45693.invoke(mongo.clj:132)"
 "driver.mongo$describe_table_field.invokeStatic(mongo.clj:131)"
 "driver.mongo$describe_table_field.invoke(mongo.clj:120)"
 "driver.mongo$describe_table$f__44945__auto____45730$iter__45731__45735$fn__45736.invoke(mongo.clj:165)"
 "driver.mongo$describe_table$f__44945__auto____45730.invoke(mongo.clj:164)"
 "driver.mongo$describe_table.invokeStatic(mongo.clj:160)"
 "driver.mongo$describe_table.invoke(mongo.clj:159)"
 "driver$fn__26496$G__26328__26505.invoke(driver.clj:51)"
 "sync.fetch_metadata$fn__35079$table_metadata__35084$fn__35085.invoke(fetch_metadata.clj:18)"
 "sync.fetch_metadata$fn__35079$table_metadata__35084.invoke(fetch_metadata.clj:15)"
 "sync.sync_metadata.fields$fn__35640$db_metadata__35645$fn__35646.invoke(fields.clj:294)"
 "sync.sync_metadata.fields$fn__35640$db_metadata__35645.invoke(fields.clj:290)"
 "sync.sync_metadata.fields$fn__35668$sync_fields_for_table_BANG___35677$fn__35680$fn__35681.invoke(fields.clj:307)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:124)"
 "sync.util$do_with_error_handling.invoke(util.clj:119)"
 "sync.sync_metadata.fields$fn__35668$sync_fields_for_table_BANG___35677$fn__35680.invoke(fields.clj:306)"
 "sync.sync_metadata.fields$fn__35668$sync_fields_for_table_BANG___35677.invoke(fields.clj:301)"
 "sync.sync_metadata.fields$fn__35706$sync_fields_BANG___35711$fn__35712.invoke(fields.clj:318)"
 "sync.sync_metadata.fields$fn__35706$sync_fields_BANG___35711.invoke(fields.clj:314)"
 "sync.sync_metadata$fn__36275$sync_db_metadata_BANG___36280$fn__36281$fn__36282.invoke(sync_metadata.clj:28)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:124)"
 "sync.util$do_with_error_handling.invoke(util.clj:119)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:122)"
 "sync.util$do_with_error_handling.invoke(util.clj:119)"
 "driver.mongo$sync_in_context$f__44945__auto____45634.invoke(mongo.clj:70)"
 "driver.mongo.util$_with_mongo_connection$fn__44938.invoke(util.clj:116)"
 "util.ssh$with_ssh_tunnel_STAR_.invokeStatic(ssh.clj:106)"
 "util.ssh$with_ssh_tunnel_STAR_.invoke(ssh.clj:91)"
 "driver.mongo.util$_with_mongo_connection.invokeStatic(util.clj:95)"
 "driver.mongo.util$_with_mongo_connection.invoke(util.clj:90)"
 "driver.mongo$sync_in_context.invokeStatic(mongo.clj:69)"
 "driver.mongo$sync_in_context.invoke(mongo.clj:68)"
 "driver$fn__26511$G__26318__26520.invoke(driver.clj:51)"
 "sync.util$sync_in_context$fn__33539.invoke(util.clj:115)"
 "sync.util$with_db_logging_disabled$fn__33536.invoke(util.clj:106)"
 "sync.util$with_start_and_finish_logging$fn__33531.invoke(util.clj:92)"
 "sync.util$with_sync_events$fn__33528.invoke(util.clj:75)"
 "sync.util$with_duplicate_ops_prevented$fn__33519.invoke(util.clj:54)"
 "sync.util$do_sync_operation.invokeStatic(util.clj:142)"
 "sync.util$do_sync_operation.invoke(util.clj:139)"
 "sync.sync_metadata$fn__36275$sync_db_metadata_BANG___36280$fn__36281.invoke(sync_metadata.clj:23)"
 "sync.sync_metadata$fn__36275$sync_db_metadata_BANG___36280.invoke(sync_metadata.clj:20)"
 "task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:36)")
```

**Comment by VikramTiwari** [comment]
@camsaul Confirming that this is only happening with Mongo based databases.

To reproduce, simply sync schema for a Mongo based database from Admin view. It might also be because I don't have any conformity in the `key:value` datatype in my mongo instance. But that non-conformance is the reason for me to use mongo, not the reason why sync should fail. If there are more than 1 conflicting datatypes, mark it as a string and move forward rather than breaking.

**Comment by jornh** [comment]
@VikramTiwari 0.30.x includes a fix to #6975. I’m wondering if that squashed this bug for you as well?

Details of the case fixed in the PR at https://github.com/metabase/metabase/pull/8072/files

Would be good if you can retest with latest release. Thanks!

**Comment by camsaul** [comment]
Duplicate of #6975

Fixed by #8072
